src/Passes/Source/Commandline/ConfigurationBindInterface.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ParameterParser.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <type_traits>
+#include <typeindex>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Interface class to bind a variable to a configuration flag. This class provides
+    /// the neccessary means to
+    class IConfigBind
+    {
+      public:
+        using String = std::string;
+
+        // Deteled constructors, deleted operators and destructor
+        //
+        IConfigBind(IConfigBind const&) = delete;
+        IConfigBind(IConfigBind&&)      = delete;
+        IConfigBind& operator=(IConfigBind const&) = delete;
+        IConfigBind& operator=(IConfigBind&&) = delete;
```

Comment:
If I understand correctly then this disables copying and moving for the derived classes. Is that what you intended? If yes, then a few words of comments about the reasons would not harm.

Reply:
Yes, this is intended. The rationale is that this class acts as a referenced variable and should not be copied or moved, but only be used in one single place. While the code would still work when allowing copy and/or move, the potential bugs that could arise from allowing this could be hard to track.

I've added a comment explaining this.